### PR TITLE
develop: remove todo

### PR DESCRIPTION
**Pull request by crazycs520**
Signed-off-by: crazycs520 <crazycs520@gmail.com>

### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by Mini256 — COMMENTED** [review]
rest LGTM

**Review by winkyao — APPROVED** [review]
LGTM

File: develop/index-best-practice.md [hunk 1]
```diff
@@ -93,7 +93,17 @@ CREATE TABLE `books` (
     SELECT * FROM books WHERE published_at >= '2022-01-01' AND published_at < '2023-01-01';
     ```
 
-    // TODO: 等表达式索引 GA 后，添加表达式索引的示例。
+    也可以使用表达式索引，例如对查询条件中的 `YEAR(published_at)` 创建一个表达式索引：
+
+    ```sql
+    CREATE INDEX published_year_idx ON books ((YEAR(published_at)));
```
[the remark this answers is not in the document]
Comment (crazycs520) [reply develop/index-best-practice.md:101]: 不能，去掉一个括号会有 SQL 语法报错。

**Review by Mini256 — APPROVED** [review]

File: develop/index-best-practice.md [hunk 2]
```diff
@@ -93,7 +93,19 @@ CREATE TABLE `books` (
...
+    ```
+
+    现在 `SELECT * FROM books WHERE YEAR(published_at)=2022;` 查询就能使用 `published_year_idx` 索引来加速查询了。
+
+    > 注意：
+    >
+    > TiDB 中，表达式索引目前是实验特性，需要在 TiDB 配置文件中开启表达式索引特性，详情可以参考 [表达式索引官方文档](https://docs.pingcap.com/zh/tidb/stable/sql-statement-create-index#%E8%A1%A8%E8%BE%BE%E5%BC%8F%E7%B4%A2%E5%BC%95)。
```
Comment (qiancai) [inline develop/index-best-practice.md:108]:
```suggestion
    > 表达式索引目前是 TiDB 的实验特性，需要在 TiDB 配置文件中开启表达式索引特性，详情可以参考 [表达式索引文档](/sql-statement-create-index.md#表达式索引)。
```

File: develop/index-best-practice.md [hunk 3]
```diff
@@ -93,7 +93,19 @@ CREATE TABLE `books` (
...
-    // TODO: 等表达式索引 GA 后，添加表达式索引的示例。
+    也可以使用表达式索引，例如对查询条件中的 `YEAR(published_at)` 创建一个表达式索引：
+
+    {{< copyable "sql" >}}
+
+    ```sql
+    CREATE INDEX published_year_idx ON books ((YEAR(published_at)));
+    ```
+
+    现在 `SELECT * FROM books WHERE YEAR(published_at)=2022;` 查询就能使用 `published_year_idx` 索引来加速查询了。
```
Comment (qiancai) [inline develop/index-best-practice.md:104]:
```suggestion
    然后通过 `SELECT * FROM books WHERE YEAR(published_at)=2022;` 查询就能使用 `published_year_idx` 索引来加速查询了。
```

File: develop/index-best-practice.md [hunk 4]
```diff
@@ -93,7 +93,19 @@ CREATE TABLE `books` (
...
+
+    > 注意：
```
Comment (qiancai) [inline develop/index-best-practice.md:106]:
```suggestion
    > **注意：**
```

**Review by qiancai — APPROVED** [review]

File: develop/index-best-practice.md [hunk 5]
```diff
@@ -93,7 +93,19 @@ CREATE TABLE `books` (
...
+
+    ```sql
+    CREATE INDEX published_year_idx ON books ((YEAR(published_at)));
+    ```
+
+    然后通过 `SELECT * FROM books WHERE YEAR(published_at)=2022;` 查询就能使用 `published_year_idx` 索引来加速查询了。
+
+    > **注意：**
+    >
+    > 表达式索引目前是 TiDB 的实验特性，需要在 TiDB 配置文件中开启表达式索引特性，详情可以参考 [表达式索引文档](/sql-statement-create-index.md#表达式索引)。
```
Comment (Icemap) [inline develop/index-best-practice.md:108]: 5.1 版本中说是实验特性：https://docs.pingcap.com/zh/tidb/v5.1/sql-statement-create-index#%E8%A1%A8%E8%BE%BE%E5%BC%8F%E7%B4%A2%E5%BC%95
在 5.2 版本中已经删除此说明：https://docs.pingcap.com/zh/tidb/v5.2/sql-statement-create-index#%E8%A1%A8%E8%BE%BE%E5%BC%8F%E7%B4%A2%E5%BC%95

当前 stable 版本为 5.4，应该是非实验特性。

Comment (Icemap) [reply develop/index-best-practice.md:108]: 说错，就是实验特性，见：
https://docs.pingcap.com/zh/tidb/stable/basic-features#%E7%B4%A2%E5%BC%95%E5%92%8C%E7%BA%A6%E6%9D%9F

但为何删除说明不得而知。

**Comment by TomShawn** [comment]
@crazycs520 @Icemap Please fix the CI link failure. Thanks~

**Comment by Icemap** [comment]
Fixed. PTAL THX. @TomShawn

**Comment by Icemap** [comment]
PTAL @shawn0915

**Comment by shawn0915** [comment]
LGTM @Icemap

**Pull request merged**